src/plugins/discover/server/saved_objects/search_migrations.ts

```diff
@@ -120,9 +132,45 @@ const migrateSearchSortToNestedArray: SavedObjectMigrationFn<any, any> = (doc) =
   };
 };
 
+/**
+ * This creates a migration map that applies search source migrations
+ */
+const getSearchSourceMigrations = (searchSourceMigrations: MigrateFunctionsObject) =>
+  mapValues<MigrateFunctionsObject, MigrateFunction>(
+    searchSourceMigrations,
+    (migrate: MigrateFunction<SerializedSearchSourceFields>): MigrateFunction =>
+      (state) => {
+        const _state = state as unknown as { attributes: SavedSearchMigrationAttributes };
```

Comment:
nit: Would it make sense to create an issue for improving `MigrateFunction` type of `kibana-utils`?

Reply:
Sure!